File: src/main/java/programmers/level3/popping_balloon/SolutionTest.java
package programmers.level3.popping_balloon;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class SolutionTest {

    @Test
    public void case1() {
        Solution solution = new Solution();
        int result = solution.solution(new int[]{9, -1, -5});
        assertThat(result).isEqualTo(3);
    }

    @Test
    public void case2() {
        Solution solution = new Solution();
        int result = solution.solution(new int[]{-16, 27, 65, -2, 58, -92, -71, -68, -61, -33});
        assertThat(result).isEqualTo(6);
    }

}

File: src/main/java/programmers/kakao2024_intern/the_most_received_gift/SolutionTest.kt
package programmers.kakao2024_intern.the_most_received_gift

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {

    val solution = Solution()

    @Test
    fun test1() {
        val friends = arrayOf("muzi", "ryan", "frodo", "neo")
        val gifts = arrayOf("muzi frodo", "muzi frodo", "ryan muzi", "ryan muzi", "ryan muzi", "frodo muzi", "frodo ryan", "neo muzi")
        assertThat(solution.solution(friends, gifts)).isEqualTo(2)
    }

    @Test
    fun test2() {
        val friends = arrayOf("joy", "brad", "alessandro", "conan", "david")
        val gifts = arrayOf("alessandro brad", "alessandro joy", "alessandro conan", "david alessandro", "alessandro david")
        assertThat(solution.solution(friends, gifts)).isEqualTo(4)
    }

    @Test
    fun test3() {
        val friends = arrayOf("a", "b", "c")
        val gifts = arrayOf("a b", "b a", "c a", "a c", "a c", "c a")
        assertThat(solution.solution(friends, gifts)).isEqualTo(0)
    }

}